drafts/dynamic-components-loading.md

```diff
@@ -0,0 +1,58 @@
+---
+title: Ditch the Template Chaos - Angular Dynamic Components Loading
+subtitle: Preventing Eye Cancer from HTML Pollution
+domain: software-engineering-corner.hashnode.dev
+tags: web-development, angular, components, html, template
+cover: https://cdn.hashnode.com/res/hashnode/image/stock/unsplash/9nXBuHKkafU/upload/32547a47488e49867018f2436a958559.jpeg?w=1600&h=840&fit=crop&crop=entropy&auto=compress,format&format=webp
+publishAs: timouti
+hideFromHashnodeCommunity: false
+ignorePost: true
+---
+
+Sometimes you have to deal with handling various views for different data types. In this article, we explore dynamic components loading in Angular.
+
+### Problem
+Let's say we are fetching a list of cameras and camera equipment for our glorious online shop. Each item is either a camera or camera equipment. You know, like tripods, lenses, bags, and whatever fancy filters come to your mind. All those items share some common attributes like price, name, some overselling description, and an ID. However, when it comes to the technical characteristics, they differ. For cameras, we want to show the list of tech attributes in a table. And for the equipment, we ought to use single values, references, or whatever fancy displaying alternative comes to your mind. The point is, that we need to employ different views depending on the item type.
+
+### Dynamic Components Loading
+Now, we already know that we need to create different view components. So, one view for the tech spec table, one for simple string lists, etc. You get the drill. The question remains how to easily switch between those view components depending on the data type of our attributes. Traditionally, you would use an `ngIf` or `ngSwitch` for such a task (or factory resolvers if you are old-school). However, when the creativity of your product manager or designer goes out of bounds, you might need to handle much more than just four view cases. In that case, both approaches don't scale well. There is enough chaos and pollution in the world. So, we want to keep at least our template nice and clean.
```

Comment:
`However, when the creativity of your product manager or designer goes out of bounds`. Maybe you could phrase this a little bit more positive?